README.md

```diff
@@ -852,6 +852,36 @@ process. To get around this, increment the current value of
 `AZUL_DEPLOYMENT_INCARNATION` in the deployment's `environment.py` file, then
 redeploy.
 
+## Caught warnings during tearDownClass
+
+When running a testcase independently in PyCharm, it's possible that the 
+`tearDownClass` fails due to an unexpected warning. If this failure occurs, add
+the warning to the list of permitted warnings found in
+[`AzulTestCase`](test/azul_test_case.py).
+
+Due to how unit tests are discovered it's possible that a warning is emitted
+when a module is loaded outside the scope of the `AzulTestCase` context manager.
+The warnings that are emitted and caught are dependent on the order in which 
+modules are loaded. For example if there was an unpermitted warning emitted in a 
+testcase located at `test/service/test_app_logging.TestServiceAppLogging`. It's 
+possible that running 
+`(cd test && python -m unittest service.test_app_logging.TestServiceAppLogging)`
+will not emit the warning within the scope of the `AzulTestCase` context 
+manager, but instead emit the warning when the unit tests are loaded by the 
+unit test discovery process. 
+By moving in to the directory that contains the test case file, 
+`(cd test/service && python -m unittest test_app_logging.TestServiceAppLogging)`
+the context manager might catch the warning and fail appropriately. This is 
+because as the part of the traversal down in to the directory the discovery 
+process attempts to load modules that it finds, this loading process might emit
+warnings that we execpted the context manager to catch. When running tests for 
+a class within PyCharm, the TestRunner attempts to be very specific in how the 
+tests are loaded, which means it's more likely that the `AzulTestCase` context 
+manager catches the warning. Within the continuous integration system unit tests
```

Comment:
You don't have to take all this verbatim, but I started rephrasing a sentence I found awkward and found a couple typos etc. on the way, so look closely for those other details, I don't think Github's dif is making them obvious.

```suggestion
the context manager might catch the warning and fail appropriately. When
running the test from a different directory, Python must traverse the directory
structure, importing modules as it goes. This loading process might emit
warnings that we otherwise would expect the context manager to catch. PyCharm's test
runner however, loads tests in a very specific order, which makes `AzulTestCase` 
context manager more likely to catch the warning.  Within the continuous integration system unit tests
```

Reply:
I added in your changes verbatim, although I'm sure Hannes will have additional input to this.